src/pubtools/_pulp/tasks/copy_repo.py

```diff
@@ -214,14 +249,23 @@ def repo_copy(copy_tasks, repo):
             tasks = list(chain.from_iterable(copy_tasks))
             return RepoCopy(tasks=tasks, repo=repo)
 
-        for src_repo, dest_repo in repo_pairs:
+        for src_repo, dest_repo in sorted(repo_pairs):
             one_pair_copies = []
+            tasks_f = f_return()
             for item in criteria or [None]:
-                tasks_f = self.pulp_client.copy_content(
-                    src_repo, dest_repo, criteria=item
+                # ensure the criterias are processed and completed/resolved in order
+                # so that non-rpm copy completes before rpm copy
+                # pylint:disable=cell-var-from-loop
+                tasks_f = f_flat_map(
```

Comment:
yeah, I think we should be fine with this pylint warning, the issue is with those variables that change each loop and are used in lambda. And if you use it like this you will got all lambdas called with values of the last iteration:
```
fcs = []
for x in range(10):
    f = lambda: x
    fcs.append(f)

for func in fcs:
    print(func())

"""
OUTPUT:
9
9
9
9
9
9
9
9
9
9
"""
```
But iiuic we should be fine because we call it immediately and test didn't discover any issue with it.